stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -732,6 +809,71 @@ end
     return matches
 end
 
+# provide completion for keyword arguments in function calls
+function complete_keyword_argument(partial, last_idx, context_module)
+    fail = Completion[], 0:-1
+
+    # Quickly abandon if the situation does not look like the completion of a kwarg
+    idx_last_punct = something(findprev(x -> ispunct(x) && x != '_' && x != '!', partial, last_idx), 0)::Int
+    idx_last_punct == 0 && return fail
+    last_punct = partial[idx_last_punct]
+    last_punct == ',' || last_punct == ';' || last_punct == '(' || return fail
+    before_last_word_start = something(findprev(in(non_identifier_chars), partial, last_idx), 0)
+    before_last_word_start == 0 && return fail
+    all(isspace, partial[nextind(partial, idx_last_punct):before_last_word_start]) || return fail
+    frange, method_name_end = find_start_brace(partial[1:idx_last_punct])
+    method_name_end ∈ frange || return fail
+
+    # At this point, we are guaranteed to be in a set of parentheses, possibly a function
+    # call, and the last word (currently being completed) has no internal dot (i.e. of the
+    # form "foo.bar") and is directly preceded by `last_punct` (one of ',', ';' or '(').
+    # Now, check that we are indeed in a function call
+    frange = first(frange):(last_punct==';' ? prevind(partial, idx_last_punct) : idx_last_punct)
+    s = replace(partial[frange], r"\!+([^=\(]+)" => s"\1") # strip preceding ! operator
+    ex = Meta.parse(s * ')', raise=false, depwarn=false)
+    isa(ex, Expr) || return fail
+    ex.head === :call || (ex.head === :. && ex.args[2] isa Expr && (ex.args[2]::Expr).head === :tuple) || return fail
```

Comment:
Could we possibly combine much of this work with `complete_methods`? Seems like some at least of this code must be duplicated and extraneous.